R/stat-function.r

```diff
@@ -49,6 +49,7 @@
 #' # Using a custom function
 #' test <- function(x) {x ^ 2 + x + 20}
 #' f + stat_function(fun = test)
+#' f + stat_function(fun = ~ .x^2 + .x + 20)
```

Comment:
Probably, `.` is better for single-argument function?

Reply:
I thought about that, too: `.x` seems to be more tidyverse consistent in that it's more referentially transparent in a piping context, but piping is not native to __ggplot2__.  I don't have a strong preference either way.

Reply:
I have a slight preference for `.x`, because I think it imposes slightly lower mental load, in particular for complex formulas. Compare:
``` r
fun = ~ (5*.^2 + 3*.) * exp(-2*.^2 + 3*. + 2)
fun = ~ (5*.x^2 + 3*.x) * exp(-2*.x^2 + 3*.x + 2)
```

Reply:
OK, I prefer `.` because it's short, but it's just my preference and I don't have strong opinion. Let's use `.x`.

(Note that in some places e.g. `sec_axis()` `.x` is not allowed, but it's just implementational details.)